packages/taquito-signer/src/derivation-tools/index.ts

```diff
@@ -0,0 +1,69 @@
+import { deriveKey } from "@stablelib/pbkdf2";
+import { SHA512 } from "@stablelib/sha512";
+import { InvalidDerivationPathError } from "../errors";
+
+export * as ECDSA from "./ecdsa";
+export * as Ed25519 from "./ed25519";
+
+export const Hard = 0x80000000;
+
+export interface ExtendedKey {
+    readonly chainCode: Uint8Array;
+    derive(index: number): ExtendedKey;
+    derivePath(path: Iterable<number>): ExtendedKey;
+}
+
+export interface ExtendedPrivateKey extends ExtendedKey {
+    derive(index: number): ExtendedPrivateKey;
+    derivePath(path: number[]): ExtendedPrivateKey;
+}
+
+// Generates the seed value from the mnemonic as specified in BIP-39.
+// The seed is the primary secret used to derive root keys and their derivatives
+export function newSeedFromMnemonic(mnemonic: string, password?: string): Uint8Array {
```

Comment:
will remove

i feel that the bip39 package would be better to use since we are using the mnemonic validator build into it so we dont have to support it ourselves